test/fixtures/mandate-fixtures.js

```diff
@@ -0,0 +1,120 @@
+'use strict'
+
+const Payer = require('../../common/classes/Payer.class')
+const Mandate = require('../../common/classes/Mandate.class')
+const GatewayAccount = require('../../common/classes/GatewayAccount.class')
+const Service = require('../../common/classes/Service.class')
+
+const randomExternalId = () => Math.random().toString(36).substring(7)
+const randomNumber = () => Math.round(Math.random() * 10000) + 1
+const randomUrl = () => 'https://' + randomExternalId() + '.com'
+
+module.exports = {
+  validTokenExchangeResponse: (opts = {}) => {
+    const data = {
+      external_id: opts.external_id || randomExternalId(),
+      mandate_reference: opts.mandate_reference || 'buy Silvia a coffee',
+      type: opts.type || 'CHARGE',
+      state: opts.state || randomNumber(),
+      return_url: opts.return_url || randomUrl(),
+      gateway_account_external_id: opts.gateway_account_external_id || randomExternalId(),
+      gateway_account_id: opts.gateway_account_id || randomNumber()
+    }
+    return {
+      getPlain: () => {
+        return data
+      }
+    }
+  },
+  validGatewayAccountResponse: (opts = {}) => {
+    const data = {
+      gatewayAccountId: opts.gatewayAccountId || randomNumber(),
+      gatewayAccountExternalId: opts.gatewayAccountExternalId || randomExternalId(),
+      paymentMethod: opts.paymentMethod || 'DIRECT_DEBIT',
+      paymentProvider: opts.paymentProvider || 'SANDBOX',
+      type: opts.type || 'TEST'
+    }
+    return {
+      getPlain: () => {
+        return data
+      }
+    }
+  },
+  validMandateResponse: (opts = {}) => {
+    const data = {
```

Comment:
is this validMandateResponse meant to represent https://github.com/alphagov/pay-direct-debit-connector/blob/master/src/main/java/uk/gov/pay/directdebit/mandate/api/GetMandateResponse.java?

Reply:
yes I think so

Reply:
Yes

Reply:
ok because the fields on GetMandateResponse are mandate_id, mandate_type, return_url, links, service_reference and mandate_reference. But there's only external_id (?) and return_url here.

Reply:
`mandate_type` is going, and we only need to put the fields in the fixture that we actually need to use out of the response. I haven't changed this, so they could be added if required by any tests